backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -212,15 +212,47 @@ public LiveOktaRepository(
           "Cannot update Okta user with unrecognized username");
     }
     User user = users.single();
+    updateUser(user, userIdentity);
+
+    return getOrganizationRoleClaimsForUser(user);
+  }
+
+  private void updateUser(User user, IdentityAttributes userIdentity) {
     user.getProfile().setFirstName(userIdentity.getFirstName());
     user.getProfile().setMiddleName(userIdentity.getMiddleName());
     user.getProfile().setLastName(userIdentity.getLastName());
     // Is it our fault we don't accommodate honorific suffix? Or Okta's fault they
     // don't have regular suffix? You decide.
     user.getProfile().setHonorificSuffix(userIdentity.getSuffix());
     user.update();
+  }
 
-    return getOrganizationRoleClaimsForUser(user);
+  public void reprovisionUser(IdentityAttributes userIdentity) {
```

Comment:
it feels like this code should be tucked into the `setUserIsActive` method that already exists -- there's an if-condition in that method for `if (active && user.getStatus() == UserStatus.SUSPENDED)`.

Reply:
I looked at that a bit, but couldn't think of a way to keep it from being confusing.  This and `setUserIsActive(false)` both expect the user to be in the `SUSPENDED` state, but diverge from there.

Reply:
you could just replace the logic under the `if (active && user.getStatus() == UserStatus.SUSPENDED)` condition and replace it with the logic you have in `reprovisionUser`. otherwise that logic in that if-condition will just dangle. but it's not a big deal either way.